src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -94,6 +94,14 @@ internal partial class Binder
 
                 foreach (var p in parameterSyntaxList.Value)
                 {
+                    if (p.AttributeLists.Count > 0 && syntax.Kind() == SyntaxKind.AnonymousMethodExpression)
```

Comment:
This is precisely the check that was happening in the parser before.  We only disallowed 'attributes' on parameters for anonymous-method-expressions.  I'm not trying to change what language is/isn't allowed by the compiler.  I'm only trying to move out *contextual* language checks out from the parser layer.

For example, because of this check you might run into an incremental parsing issue.  i.e. a parameter that was parsed with an attribute, for some non-anonymous-method-expression might get reused by an anonymous-method-expression.  But because it was reused in this new context, we would not then produce the parse error.  This would violate our incremental parsing invariants.

So, to avoid the issue entirely, it's useful to just get rid of contextual parsing as much as possible in the parser.